File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() == 0){
            return 0;
        } else if(nums.size() == 1){
            return 1;
        }

        sort(nums.begin(), nums.end());

        int res = 0;
        int curr = nums[0];
        int max = 0;
        int i = 0;

        while(i < nums.size()){
            if(curr != nums[i]){
                curr = nums[i];
                max = 0;
            }
            while(i < nums.size() && nums[i] == curr){
                i++;
            }
            max++;
            curr++;
            if(res < max){
                res = max;
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        self.res = []
        self.backtrack([], nums, [False] * len(nums))
        return self.res

    def backtrack(self, curr: List[int], nums: List[int], picked: List[bool]):
        if len(curr) == len(nums):
            self.res.append(curr.copy())
            return

        for i in range(len(nums)):
            if not picked[i]:
                curr.append(nums[i])
                picked[i] = True
                self.backtrack(curr, nums, picked)
                curr.pop()
                picked[i] = False
